dropboxhandler/dropboxhandler.py

```diff
@@ -61,6 +63,24 @@ def extract_barcode(path):
         raise ValueError("more than one barcode in filename")
     return barcodes[0]
 
+def check_not_empty(path):
+    """Check the size of the incoming file or folder. Throws error
+    if it's empty.
+    """
+    size = fstools.get_size(path)
+    if size == 0:
+        raise ValueError('%s is empty.' % path)
+
+def check_not_hidden(path):
+    """If file is hidden (starts with '.') throw error. To be sure
+    the file processed by datamover started with '.', we check for
+    the timestamp before it, as well.
+    Possible names dependent on processing status:
+    <timestamp>_._<rest of name>
+    <barcode>_<timestamp>_._<rest of name>
+    """
+    if re.match(".*"+TIMESTAMP_REGEX+"_\.", path):
+        raise ValueError('%s is a hidden file.' % path)
```

Comment:
SRP violation of functions. Imo the functions should just check if a file is empty and return a boolean (following Command & Query separation).

What about 
```suggestion
def isEmpty(path):
    """Check the size of the incoming file or folder. Throws error
    if it's empty.
    """
    return fstools.get_size(path) == 0
  
def isHidden(path):
    """If file is hidden (starts with '.') throw error. To be sure
    the file processed by datamover started with '.', we check for
    the timestamp before it, as well.
    Possible names dependent on processing status:
    <timestamp>_._<rest of name>
    <barcode>_<timestamp>_._<rest of name>
    """
    return os.path.basename(os.path.abspath(path)).startswith('.')
    
if (isHidden(path) or isEmpty(path)): 
   # Do reporting strategy here
```

Reply:
Though this solution is not robust for different OSs and FSs, should be enough for our case